src/Admin/AbstractAdmin.php

```diff
@@ -2612,10 +2644,22 @@ public function supportsPreviewMode()
     }
 
     /**
+     * NEXT_MAJOR: Remove this.
+     *
+     * @deprecated since since sonata-project/sonata-admin-bundle 3.x, to be remove in 4.0.
+     *
      * Set custom per page options.
      */
     public function setPerPageOptions(array $options)
     {
+        @trigger_error(
+            sprintf(
+                'The method %s is deprecated since sonata-project/sonata-admin-bundle 3.x and will be removed in 4.0.',
+                __METHOD__
+            ),
+            E_USER_DEPRECATED
+        );
+
```

Comment:
```suggestion
        @trigger_error(sprintf(
            'The method %s is deprecated since sonata-project/sonata-admin-bundle 3.x and will be removed in 4.0.',
            __METHOD__
        ), E_USER_DEPRECATED);
```